src/ApiResponse/SearchReportApiResponse.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Silktide\ProspectClient\ApiResponse;
+
+use Countable;
+use Iterator;
+use Silktide\ProspectClient\Entity\Report;
+
+class SearchReportApiResponse extends AbstractApiResponse implements Countable, Iterator
```

Comment:
Making this an iterator muddles the purpose of the class a little.

It's no longer a pure Response to your Request, it's now the response (encapsulating http codes etc) as well as essentially being an array for the underlying reports?

It's easy to get enthusiastic about iterators, but unless you're:
a. Creating a formal collection class or
b. Dealing with something that has performance considerations (normally something that requires buffering, such as large database queries that involve cursors)

It's not normally the right use case. 

In this case, you'd be better off just hydrating an array of `Report` classes straight away and returning a `getReports()`

Reply:
Roger that.